src/Core/Stock/StockManager.php

```diff
@@ -174,9 +174,99 @@ public function updateQuantity($product, $id_product_attribute, $delta_quantity,
             )
         );
 
+        if ($this->checkIfMustSendLowStockAlert($product, $id_product_attribute, $stockAvailable->quantity)) {
+            $this->sendLowStockAlert($product, $id_product_attribute, $stockAvailable->quantity);
+        }
+
         $cacheManager->clean('StockAvailable::getQuantityAvailableByProduct_'.(int)$product->id.'*');
     }
 
+    protected function checkIfMustSendLowStockAlert($product, $id_product_attribute, $newQuantity)
+    {
+        if (!\Configuration::get('PS_STOCK_MANAGEMENT')) {
+            return false;
+        }
+
+        // Do not send mail if multiples product are created / imported.
+        if (defined('PS_MASS_PRODUCT_CREATION')) {
+            return false;
+        }
+
+        $product_has_attributes = $product->hasAttributes();
+        if ($product_has_attributes && $id_product_attribute) {
+            $combination = new \Combination($id_product_attribute);
+            if ($combination->low_stock_alert
+                && (int) $combination->low_stock_threshold > 0
+                && $newQuantity <= (int) $combination->low_stock_threshold
+            ) {
+                return true;
+            }
+        } elseif ($product_has_attributes && !$id_product_attribute) {
+            if ($product->low_stock_alert
+                && (int) $product->low_stock_threshold > 0
+                && $newQuantity <= (int) $product->low_stock_threshold
+            ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    protected function sendLowStockAlert($product, $id_product_attribute, $newQuantity)
+    {
+        $context = \Context::getContext();
+        $id_shop = (int) $context->shop->id;
+        $id_lang = (int) $context->language->id;
+        $configuration = \Configuration::getMultiple(
+            array(
+                'MA_LAST_QTIES',
+                'PS_STOCK_MANAGEMENT',
+                'PS_SHOP_EMAIL',
+                'PS_SHOP_NAME',
+            ), null, null, $id_shop
+        );
```

Comment:
This method call does not follow the[ PSR requirements](http://www.php-fig.org/psr/psr-2/#46-method-and-function-calls) (one parameter per line).